Content.Server/Singularity/Components/EmitterComponent.cs

```diff
@@ -18,14 +20,102 @@ public sealed class EmitterComponent : Component
         public const float Volume = 0.5f;
         public const float Distance = 6f;
 
-        [ViewVariables] public int FireShotCounter;
+        /// <summary>
+        /// counts the number of consecutive shots fired.
+        /// </summary>
+        [ViewVariables]
+        public int FireShotCounter;
 
-        [ViewVariables] [DataField("fireSound")] public SoundSpecifier FireSound = new SoundPathSpecifier("/Audio/Weapons/emitter.ogg");
-        [ViewVariables] [DataField("boltType")] public string BoltType = "EmitterBolt";
-        [ViewVariables] [DataField("powerUseActive")] public int PowerUseActive = 500;
-        [ViewVariables] [DataField("fireBurstSize")] public int FireBurstSize = 3;
-        [ViewVariables] [DataField("fireInterval")] public TimeSpan FireInterval = TimeSpan.FromSeconds(2);
-        [ViewVariables] [DataField("fireBurstDelayMin")] public TimeSpan FireBurstDelayMin = TimeSpan.FromSeconds(2);
-        [ViewVariables] [DataField("fireBurstDelayMax")] public TimeSpan FireBurstDelayMax = TimeSpan.FromSeconds(10);
+        [DataField("fireSound"), ViewVariables]
+        public SoundSpecifier FireSound = new SoundPathSpecifier("/Audio/Weapons/emitter.ogg");
+
+        /// <summary>
+        /// The entity that is spawned when the emitter fires.
+        /// </summary>
+        [DataField("boltType"), ViewVariables]
+        public string BoltType = "EmitterBolt";
+
+        /// <summary>
+        /// The current amount of power being used.
+        /// </summary>
+        [DataField("powerUseActive"), ViewVariables]
+        public int PowerUseActive = 600;
+
+        /// <summary>
+        /// The base amount of power that is consumed.
+        /// Used in machine part rating calculations.
+        /// </summary>
+        [DataField("basePowerUseActive"), ViewVariables(VVAccess.ReadWrite)]
+        public int BasePowerUseActive = 600;
+
+        /// <summary>
+        /// Multiplier that is applied to the basePowerUseActive
+        /// to get the actual power use.
+        /// </summary>
+        [DataField("powerUseMultiplier")]
+        public float PowerUseMultiplier = 0.75f;
+
+        /// <summary>
+        /// The machine part used to reduce the power use of the machine.
+        /// </summary>
+        [DataField("machinePartPowerUse", customTypeSerializer: typeof(PrototypeIdSerializer<MachinePartPrototype>))]
+        public string MachinePartPowerUse = "Capacitor";
```

Comment:
Maybe this should be its own component/system, like how #11662 did it for capacity? That way we have less code duplication.

Reply:
Actually I see you have a pr for that, #11777.

Reply:
the only reason i'm not doing this in a separate system is because currently emitter code is kinda shit and directly sets the power intake value. I suppose it should probably just be refactored, but i decided to just deal with it here and leave refactoring for another time.